packages/onchain/src/orderbook/orderbook.cairo

```diff
@@ -0,0 +1,100 @@
+#[starknet::interface]
+pub trait IOrderbook<TContractState> {
+    fn request_inscription(
+        ref self: TContractState, 
+        inscription_data: ByteArray, 
+        satoshi: felt252, 
+        currency_fee: felt252, 
+        submitter_fee: felt252
+    ) -> u16;
+    fn cancel_inscription(ref self: TContractState, inscription_id: u16);
+    fn lock_inscription(ref self: TContractState, inscription_id: u16);
+    fn submit_inscription(ref self: TContractState, inscription_id: u16);
+    fn query_inscription(self: @TContractState, inscription_id: u16) -> (ByteArray, felt252);
+}
+
+#[starknet::contract]
+mod Orderbook {
+    use core::byte_array::ByteArray;
+    use starknet::storage::{ 
+        Map, StorageMapReadAccess, StorageMapWriteAccess, StoragePathEntry, 
+        StoragePointerReadAccess, StoragePointerWriteAccess, 
+    };
+    use starknet::{ContractAddress, get_caller_address};
+
+    #[storage]
+    struct Storage {
+        // ID of the next inscription.
+        new_inscription_id: u16,
```

Comment:
I think its better to use the `u32` (size type) for `id`s, since it can be more naturally compared to lengths in arrays.